cargo-rbmt/src/prerelease.rs

```diff
@@ -0,0 +1,180 @@
+
+        if config.skip {
+            skipped.push(crate_dir.as_str());
+            quiet_println(&format!("Skipping crate: {} (marked as skip)", crate_dir));
+            continue;
+        }
+
+        quiet_println(&format!("Checking crate: {}", crate_dir));
+
+        let _dir = sh.push_dir(crate_dir);
+
+        // Run all pre-release checks. Return immediately on first failure.
+        if let Err(e) = check_todos(sh) {
+            eprintln!("Pre-release check failed for {}: {}", crate_dir, e);
+            return Err(e);
+        }
+
+        if let Err(e) = check_crate(sh) {
+            eprintln!("Pre-release check failed for {}: {}", crate_dir, e);
+            return Err(e);
+        }
+    }
+
+    quiet_println("All pre-release checks passed");
+    Ok(())
+}
+
+/// Check for TODO comments in source files.
+fn check_todos(sh: &Shell) -> Result<(), Box<dyn std::error::Error>> {
+    quiet_println("Checking for TODO comments...");
+
+    let mut todos = Vec::new();
+    let src_dir = sh.current_dir().join("src");
+
+    // Recursively walk the src/ directory.
+    let mut dirs_to_visit = vec![src_dir];
+    while let Some(dir) = dirs_to_visit.pop() {
+        for entry in fs::read_dir(dir)? {
+            let entry = entry?;
+            let path = entry.path();
+
+            if path.is_dir() {
+                dirs_to_visit.push(path);
+            } else if path.extension().and_then(|s| s.to_str()) == Some("rs") {
+                // Check Rust source files for TODO comments.
+                let file = fs::File::open(&path)?;
+                let reader = BufReader::new(file);
+
+                for (line_num, line) in reader.lines().enumerate() {
+                    let line = line?;
+                    if line.contains("// TODO") || line.contains("/* TODO") {
+                        todos.push((path.clone(), line_num + 1, line));
+                    }
+                }
+            }
+        }
+    }
+
+    if !todos.is_empty() {
+        eprintln!("\nFound {} TODO comment(s):", todos.len());
+        for (file, line_num, line) in &todos {
+            eprintln!("{}:{}:{}", file.display(), line_num, line.trim());
+        }
+        return Err(format!("Found {} TODO comments", todos.len()).into());
+    }
+
+    quiet_println("No TODO comments found");
+    Ok(())
+}
+
+/// Check that the crate can be packaged and published.
+///
+/// A crate may work with local path dependencies, but fail when published
+/// because the version specifications don't match the published versions
+/// or don't resolve correctly.
+fn check_crate(sh: &Shell) -> Result<(), Box<dyn std::error::Error>> {
+    quiet_println("Running cargo package...");
+    quiet_cmd!(sh, "cargo package").run()?;
+    quiet_println("cargo package succeeded");
+
+    let (package_name, version) = get_package_info(sh)?;
+    let package_dir = format!("target/package/{}-{}", package_name, version);
```

Comment:
I have `cargo` configured to use a system-wide target directory so this won't work for me. Can we use `CARGO_TARGET_DIR` or something like that instead?

Reply:
Oh yea good thinking. Let me pull this one into draft and harden it up a bit.